activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -37,7 +37,7 @@ def self.add_counter_cache_callbacks(model, reflection)
       }
 
       klass = reflection.class_name.safe_constantize
-      klass._counter_cache_columns << cache_column if klass && klass.respond_to?(:_counter_cache_columns)
+      klass._counter_cache_columns |= [cache_column] if klass && klass.respond_to?(:_counter_cache_columns)
```

Comment:
@fatkodima Just curious to know, how this exactly fixes the issue. In both cases, we are appending the column to the `_counter_cache_columns` right?

Reply:
For `<<` we modify the original object, but for `|=` we create a new object with the original contents and the new column and assign to the attribute (original object is intact).